test/e2e/framework/framework.go

```diff
@@ -115,23 +118,17 @@ func init() {
 	// Create a directory (if not exists) $HOME/.tanzu-cli-e2e/temp
 	_ = CreateDir(FullPathForTempDir)
 
-	// TODO:cpamuluri: need to move plugins info to configuration file with positive and negative use cases - github issue: https://github.com/vmware-tanzu/tanzu-cli/issues/122
-	PluginsForLifeCycleTests = []*PluginInfo{
-		{Name: "cluster", Target: "kubernetes", Version: "v9.9.9", Description: "cluster functionality"},
-		{Name: "cluster", Target: "mission-control", Version: "v9.9.9", Description: "cluster functionality"},
-		{Name: "pinniped-auth", Target: "global", Version: "v9.9.9", Description: "pinniped-auth functionality"},
-		{Name: "telemetry", Target: "global", Version: "v9.9.9", Description: "telemetry functionality"},
-		{Name: "telemetry", Target: "kubernetes", Version: "v9.9.9", Description: "telemetry functionality"},
-		{Name: "plugin-with-sha", Target: "global", Version: "v9.9.9", Description: "plugin-with-sha functionality"},
+	inputConfPath := os.Getenv(TanzuCliE2EInputConfigDataFilePath)
+	if inputConfPath == "" {
+		pwd, _ := os.Getwd()
+		inputConfPath = strings.Split(pwd, "test/e2e")[0] + DefaultInputConfigDataFilePath
 	}
-
-	// TODO:cpamuluri: need to move Plugin Groups to configuration file with positive and negative use cases - github issue: https://github.com/vmware-tanzu/tanzu-cli/issues/122
-	PluginGroupsForLifeCycleTests = []*PluginGroup{
-		{Group: "vmware-tkg/default", Latest: "v9.9.9", Description: "Desc for vmware-tkg/default:v9.9.9"},
-		{Group: "vmware-tmc/tmc-user", Latest: "v9.9.9", Description: "Desc for vmware-tmc/tmc-user:v9.9.9"},
-		{Group: "vmware-tanzucli/essentials", Latest: "v9.9.9", Description: "Desc for vmware-tanzucli/essentials:v9.9.9"},
-		{Group: "vmware-tkg/shortversion", Latest: "v1", Description: "Desc for vmware-tkg/default:v1"},
-		{Group: "vmware-tkg/shortversion", Latest: "v1.1", Description: "Desc for vmware-tkg/default:v1"},
+	log.Infof("input config data file path: %v", inputConfPath)
+	confData, err := ReadAndProcessFile[InputConfigData](inputConfPath)
+	if err != nil {
+		log.Fatal(err, "failed to read input config data file")
 	}
-	EssentialPluginGroups = []*PluginGroup{{Group: "vmware-tanzucli/essentials", Latest: "v9.9.9", Description: "Desc for vmware-tanzucli/essentials:v9.9.9"}}
+	PluginsForLifeCycleTests = confData.PluginsForLifeCycleTests
+	PluginGroupsForLifeCycleTests = confData.PluginGroupsForLifeCycleTests
+	EssentialPluginGroups = confData.EssentialPluginGroups
```

Comment:
This is not specific to the Framework I feel. Can we move this code out of Framework under the Tanzu CLI Core Tests?